imap/ctl_mboxlist.c

```diff
@@ -1007,6 +1007,9 @@ int main(int argc, char *argv[])
 
     case M_POPULATE:
         syslog(LOG_NOTICE, "%spopulating mupdate", warn_only ? "test " : "");
+#if defined __GNUC__ &&  __GNUC__ > 6
```

Comment:
I did implement it as a `#define`  initially (you can see that I've left the `GCC_VERSION` macro in `util.h` that I've still left there). But I wasn't very comfortable with the macro way, since they are an absolute pain when debugging.
Then again, if you think that it is better, I shall go ahead an update the patch.